core/services/relay/evm/mercury/wsrpc/client.go

```diff
@@ -115,9 +122,15 @@ func newClient(lggr logger.Logger, clientPrivKey csakey.KeyV2, serverPubKey []by
 	}
 }
 
-func (w *client) Start(_ context.Context) error {
-	return w.StartOnce("WSRPC Client", func() error {
-		if err := w.dial(context.Background()); err != nil {
+func (w *client) Start(ctx context.Context) error {
+	return w.StartOnce("WSRPC Client", func() (err error) {
+		// NOTE: This is not a mistake, dial is non-blocking so it should use a
+		// background context, not the Start context
+		if err = w.dial(context.Background()); err != nil {
```

Comment:
```suggestion
		if err = w.dial(w.chStop.Ctx()); err != nil {
```

Reply:
Oh but that returns a cancel func too so it doesn't exactly fit. Maybe nbd :shrug:

Reply:
You could set the cancel func on the `*client` struct and call it during close, that might work?

Reply:
Maybe but then it seems kind of redundant :shrug: